tests/cc/py_cc_toolchain/py_cc_toolchain_tests.bzl

```diff
@@ -0,0 +1,85 @@
+# Copyright 2023 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for py_cc_toolchain."""
+
+load("@rules_testing//lib:analysis_test.bzl", "analysis_test", "test_suite")
+load("@rules_testing//lib:truth.bzl", "matching")
+load("//tests:cc_info_subject.bzl", "cc_info_subject")
+load("//tests:default_info_subject.bzl", "default_info_subject")
+load("//tests:py_cc_toolchain_info_subject.bzl", "PyCcToolchainInfoSubject")
+
+_tests = []
+
+def _py_cc_toolchain_test(name):
+    analysis_test(
+        name = name,
+        impl = _py_cc_toolchain_test_impl,
+        target = "//tests/cc:fake_py_cc_toolchain_impl",
+        attrs = {
+            "header": attr.label(
+                default = "//tests/cc:fake_header.h",
+                allow_single_file = True,
+            ),
+        },
+    )
+
+def _py_cc_toolchain_test_impl(env, target):
+    env.expect.that_target(target).has_provider(platform_common.ToolchainInfo)
+
+    toolchain = PyCcToolchainInfoSubject.new(
+        target[platform_common.ToolchainInfo].py_cc_toolchain,
+        meta = env.expect.meta.derive(expr = "py_cc_toolchain_info"),
+    )
+    toolchain.python_version().equals("3.999")
+
+    toolchain_headers = toolchain.headers()
+    toolchain_headers.providers_map().keys().contains_exactly(["CcInfo", "DefaultInfo"])
+
+    cc_info = cc_info_subject(
+        # TODO: Use DictSubject.get once available,
+        # https://github.com/bazelbuild/rules_testing/issues/51
```

Comment:
Can we file an issue to track this TODO?

Reply:
Filed https://github.com/bazelbuild/rules_python/issues/1297